src/provider/hexadecimal/Hexadecimal.tsx

```diff
@@ -22,18 +22,23 @@ class ProviderHandlers {
 
 const providerHandler: ProviderHandlers = new ProviderHandlers();
 
-function HexadecimalProvider({children}: IHexadecimalProviderProps) {
+function HexadecimalProvider({ children }: IHexadecimalProviderProps) {
     const [ hexColor, setHexColor ] = useState<string>("");
 
+    const deferredColor = useDeferredValue(hexColor);
+
     useEffect(() => {
         providerHandler.verifyColorEntered(setHexColor);
     },[]);
     
-
-    const values:IValuesProvider = {
-        hexColor,
+    const values: IValuesProvider = {
+        hexColor: deferredColor,
         rgbColor: "",
-        setHexColor: (hex:string) => debounce({callback: () => setHexColor(hex)})
+        setHexColor: (hex:string, useDebounce?: boolean) => {
```

Comment:
The `debounce` implementation is weird-ish. There's one unique timeout shared across the entire app, meaning that if I was using `debounce` for two separate callbacks they would cancel each other without allowing both to end up running... it's fine as it is as of now, though, but I'd go with an approach that scales better.

Reply:
It shouldn't cause problems, depending on how it's used. For example, if you use it with buttons, like the button color change that alters values very quickly, it could affect other components that rely on the value changing to trigger another debounce. This would cause the execution of one debounce to chain the execution of others unintentionally. However, in the context of the app, I made sure to address that issue. You're right that if someone else who is unaware of this detail uses it, it could cause that problem in the future. In any case, the debounce

When calling a function in a component, the memory space is not the same, so it wouldn’t affect the debounce of another. In the case of setHexColor, it would affect it, but using the debounce function independently wouldn’t.

Reply:
Here's [a sandbox environment proving the point I'm trying to make](https://codesandbox.io/p/sandbox/xcw9f5).

I have two separate buttons with two separate functions. I want both to be debounced independently, but each of them affects the other since the timer is shared across the entire app, given it's declared at module level.

Reply:
Your point is that a debounce cancels the previous debounce, and you are right. However, in the app, this is not an important issue since the debounce is triggered by the color picker and the generate random button.

This problem can be resolved by adding the timeout inside the function or creating a class

Reply:
Yup, agreed.